src/Symfony/Component/Cache/composer.json

```diff
@@ -30,7 +30,8 @@
         "cache/integration-tests": "dev-master",
         "doctrine/cache": "~1.6",
         "doctrine/dbal": "~2.4",
-        "predis/predis": "~1.0"
+        "predis/predis": "~1.0",
+        "symfony/dsn": "3.4"
```

Comment:
`symfony/dsn` is a requirement to keep BC. But in sf 4.0 in could be a suggesiton.

What is the upgrade path to deprecate a requirement?

Reply:
We have none: just read UPRGRADE files, that's what we did in the past